src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -121,6 +131,10 @@ object Task {
     // Regular expression for matching taskIds before instance-era
     private val LegacyTaskIdRegex = """^(.+)[\._]([^_\.]+)$""".r
 
+    // Regular expression for matching resident app taskIds
+    private val ResidentTaskIdRegex = """^(.+)([\._])([^_\.]+)(\.\#)(\d+)$""".r
+    private val ResidentTaskIdAttemptSeparator = ".#"
```

Comment:
wondering if there's any impact to things that might use taskID in http query params since the `#` has special meaning in http URLs (for web browsers)